### PR TITLE
only allow for includeState aggregate queries when assessment type is…

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
… summative

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/ReportQuery.java [hunk 1]
```diff
@@ -184,7 +185,7 @@ public ReportQuery build() {
             query.subjectCodes = subjectCodes;
             query.schoolYears = schoolYears;
             query.assessmentGradeCodes = assessmentGradeCodes;
-            query.includeState = includeState;
+            query.includeState = includeState && SUMMATIVE.code().equals(assessmentTypeCode);
```
Comment (agorina) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/model/ReportQuery.java]: I think our philosophy is that our builders do not have any business logic. 
I would put this check into `checkIsValid` method and make sure UI never calls it with the invalid parameters.

**Comment by agorina** [comment]
Is it correct to assume that UI change is coming too?

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/ReportQuery.java (hunk 1 above)
Comment (malaffoon) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/model/ReportQuery.java]: 👍

**Comment by dsnyder-fairway** [comment]
I didn't think UI changes was needed for this, since we the checkbox states its for summative only. My idea was that we would ignore the box being checked and default it to false in the cases where "Include State results (Summative only)" was checked for non-summative aggregate reports.

**Pull request closed without merging**